### PR TITLE
CI Update to Add MSBuild 17

### PR DESCRIPTION
**Pull request by v-alje**
Build.Reason:Manual by Alma Jenks
 Build.Url:https://apidrop.visualstudio.com/Content%20CI/_build/results?buildId=267835&view=results
 source_repo.branch:release-msbuild
 source_repo.url:https://apidrop.visualstudio.com/_git/binaries

## Summary

Describe your changes here.

Fixes #Issue_Number (if available)

### CODE REVIEW

**Review by gewarren — COMMENTED** [review]
I didn't go through the whole thing, but I'm wondering why there are so many changes unrelated to msbuild. I see you added some System.* binaries to the dependencies folder in this commit: https://apidrop.visualstudio.com/binaries/_git/binaries/commit/8e5074feabf8d7a7139946a368eb9680715c68b0?refName=refs/heads/release-msbuild. But I didn't think we were generating ECMAXML for dependencies. Thoughts?

File: xml/Microsoft.Extensions.Logging/LogLevel.xml
Comment (eerhardt) [inline xml/Microsoft.Extensions.Logging/LogLevel.xml:28]: These changes don't seem related to MSBuild 17. Can you revert them?

**Comment by v-alje** [comment]
@gewarren I have reduced this to just the MSBuild changes.   

The unexpected changes don't seem to be coming from the dependencies.  The extra API are not in any of those dll files.

There are a few (System.Formats.Cbor, System.IO.Compression.Brotli, and System.Numerics.Vectors) which might be explained by the XML being in the binaries repo alongside the dlls. If the ECMA XML for those had been manually modified, they would revert.

The rest I can only presume have to do with any changes made to the CI system since the last run, or differences to the generation system when Auto-Import is checked.

File: xml/Microsoft.Build.Construction/ProjectElement.xml
Comment (gewarren) [inline xml/Microsoft.Build.Construction/ProjectElement.xml:598]:
```suggestion
            Hook for subclasses to specify whether the given <paramref name="attribute" /> should be cloned or not.
```

File: xml/Microsoft.Build.Framework/IBuildEngine10.xml
Comment (gewarren) [inline xml/Microsoft.Build.Framework/IBuildEngine10.xml:44]:
```suggestion
```

File: xml/Microsoft.Build.Framework/SdkReference.xml
Comment (gewarren) [inline xml/Microsoft.Build.Framework/SdkReference.xml:82]: Losing some valuable comments here. Can you check if this is happening anywhere else?

File: xml/Microsoft.Build.Framework/SdkResolver.xml
Comment (gewarren) [inline xml/Microsoft.Build.Framework/SdkResolver.xml:120]: This file should be reverted.

File: xml/Microsoft.Build.Framework/SdkResult.xml
Comment (gewarren) [inline xml/Microsoft.Build.Framework/SdkResult.xml:19]: This file should be reverted.

File: xml/Microsoft.Build.Logging/ProfilerLogger.xml
Comment (gewarren) [inline xml/Microsoft.Build.Logging/ProfilerLogger.xml:95]: This file should be reverted. The original has better formatting.

**Comment by gewarren** [comment]
Also, can you address the build warnings?

It's also probably a good idea to use the [DocsPortingTool](https://github.com/carlossanlop/DocsPortingTool) to backport any doc comment changes that have been made directly to this repo for MSBuild docs.

**Comment by v-alje** [comment]
@ghogen can you address @gewarren's comments?

**Comment by ghogen** [comment]
@v-alje @gewarren 
At some point, we changed the source-of-truth for MSBuild docs from the dotnet-docs-api repo to the MSBuild sources, so if I understand correctly, the DocsPortingTool wouldn't be applicable, because we're already pulling in the docs from MSBuild sources. Also, I think any changes should be made directly in MSBuild sources, but we wouldn't see the effect of those changes until the next version of MSBuild.  Does it make sense to fix the warnings and accept suggestions, knowing those are only temporary and will be overwritten later when we import the next round of comments from MSBuild?

**Comment by ghogen** [comment]
I would guess we make the changes in both the docs repo and in MSBuild sources, so that we unblock the PR and also update the MSBuild sources.

**Comment by ghogen** [comment]
@v-alje @gewarren 
The following PR should address the warnings. https://github.com/dotnet/dotnet-api-docs/pull/7482
If one of you could review and merge that and also commit Genevieve's suggestions above (I don't have permissions to accept the changes), I would appreciate that.

File: xml/Microsoft.Build.Framework/SdkReference.xml
Comment (ghogen) [reply xml/Microsoft.Build.Framework/SdkReference.xml:82]: So far I've determined that MSBuild sources never had those comments, so this is one case where probably, the docs XML files had content, but MSBuild sources did not.

**Comment by gewarren** [comment]
Would it be okay to rerun the CI without importing comments? I don't think I saw any useful comments being imported. Only useful comments being overwritten and pretty formatting being unprettified.

**Comment by gewarren** [comment]
> @v-alje @gewarren At some point, we changed the source-of-truth for MSBuild docs from the dotnet-docs-api repo to the MSBuild sources, so if I understand correctly, the DocsPortingTool wouldn't be applicable, because we're already pulling in the docs from MSBuild sources. Also, I think any changes should be made directly in MSBuild sources, but we wouldn't see the effect of those changes until the next version of MSBuild. Does it make sense to fix the warnings and accept suggestions, knowing those are only temporary and will be overwritten later when we import the next round of comments from MSBuild?

The DocsPortingTool works in both directions, so I was suggesting we port comments from dotnet-api-docs to the source code. I really hate to see nice comments going to waste.

**Comment by v-alje** [comment]
@gewarren Sure. We can give it a try. I have kicked off a new CI run. It will push to the smoke-test-no-import branch.  

There is one new namespace we will need to look at once it is done.

**Comment by ghogen** [comment]
@v-alje @gewarren 
We've previously agreed that the source-of-truth for MSBuild docs would be the MSBuild source, so I don't think we can just arbitrarily change that.  I think the DocsPortingTool sounds promising. I don't really know yet what it does, though, but I also don't want to lose any good comment work, so getting them ported back into MSBuild makes sense, as well as fixing up some of the files manually for the current PR.

**Comment by ghogen** [comment]
For the one additional warning, that's fixed in https://github.com/dotnet/dotnet-api-docs/pull/7483
Let's not merge yet, though, we still need to handle the comments that are being lost.

**Comment by gewarren** [comment]
> Let's not merge this one yet, though, we still need to handle the comments that are being lost.

I propose for this import, we don't import any comments, that way we won't overwrite any. That way we can get the APIs updated for MSBuild 17 now, and then take the time to backport any comments to the dotnet/msbuild repo before reimporting *with* doc comments (if necessary).

File: xml/Microsoft.Build.Logging/BinaryLogger.xml
Comment (ghogen) [inline xml/Microsoft.Build.Logging/BinaryLogger.xml:95]:
```suggestion
            Initializes the logger by subscribing to events of the specified event source.
```

File: xml/Microsoft.Build.Tasks/CreateItem.xml
Comment (ghogen) [inline xml/Microsoft.Build.Tasks/CreateItem.xml:116]: Here's an example of where the imported comments from source have better formatting, that we wouldn't get from just importing w/o comments.

File: xml/Microsoft.Build.Tasks/CreateProperty.xml
Comment (ghogen) [inline xml/Microsoft.Build.Tasks/CreateProperty.xml:164]: Another improvement we would lose if we just import without comments.

**Comment by ghogen** [comment]
This PR resolves the comment improvements that were overwritten. It also can be used as a guide for backporting to the MSBuild source. https://github.com/dotnet/dotnet-api-docs/pull/7484

**Comment by ghogen** [comment]
@gewarren 
I went through meticulously and captured all the comments that were overwritten and made sure they were preserved. So at this point, no comments are being lost.  I also have a clear guide to backport those into MSBuild sources. We have to import comments from source for these builds, because that's how we agreed with the MSBuild dev team that this process would be going forward, and I do see that they have some comment improvements that should also be published.

I should have time next week to backport the comments into MSBuild so the work on reconciling these won't be required ever again.  However, any new comment changes made by your team would still need to handled in the future.

**Comment by gewarren** [comment]
> I went through meticulously and captured all the comments

Thank you for doing that!

**Comment by v-alje** [comment]
The CI job without comment import completed. The pull request is: #7485

**Review by gewarren — APPROVED** [review]

**Pull request merged**
